catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java

```diff
@@ -223,4 +247,53 @@ private String buildOverviewTitle(String title) {
         outputStream.close();
         return thumbnailBytes;
     }
+
+    private void addDerivedResourceAttribute(Metacard metacard, ContentItem contentItem) {
+        Attribute attribute = metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+        if (attribute == null) {
+            attribute = new AttributeImpl(Metacard.DERIVED_RESOURCE_URI, contentItem.getUri());
+        } else {
+            AttributeImpl newAttribute = new AttributeImpl(attribute);
+            newAttribute.addValue(contentItem.getUri());
+            attribute = newAttribute;
+        }
+
+        metacard.setAttribute(attribute);
+    }
+
+    private int calculateOverviewHeight(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            int newHeight = (int) Math.round(((height * (maxSideLength / width))));
+            return newHeight;
+        }
+        return Math.min(height, (int) maxSideLength);
+    }
+
+    private int calculateOverviewWidth(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            return Math.min(width, (int) maxSideLength);
+        }
+
+        int newWidth = (int) Math.round(width * (maxSideLength / height));
+        return newWidth;
+    }
+
+    public void setMaxSideLength(int maxSideLength) {
+        if (maxSideLength > 0) {
+            this.maxSideLength = maxSideLength;
+        } else {
+            LOGGER.warn(
```

Comment:
Keeping this log level at `warn` since this is a startup issue only where the blueprint config would need to be updated with a valid value.